Bluepill-runner/Bluepill-runner/BPPacker.m

```diff
@@ -32,34 +31,28 @@ + (NSMutableArray *)packTests:(NSArray *)xcTestFiles withNoSplitList:(NSArray *)
         }
         return NULL;
     }
+    NSMutableDictionary *testsToRunByTestFilePath = [[NSMutableDictionary alloc] init];
+    NSUInteger totalTests = 0;
     for (BPXCTestFile *xctFile in sortedXCTestFiles) {
         if (![noSplit containsObject:[xctFile name]]) {
-            totalTests += [xctFile numTests];
+            NSMutableSet *testsToRun = [[NSMutableSet alloc] initWithArray:[xctFile allTestCases]];
+            if (testCasesToRun) {
+                [testsToRun intersectSet:[[NSSet alloc] initWithArray:testCasesToRun]];
```

Comment:
Yeah I'll rename it to be more clear. Really `testsToRun` is only for this bundle, so I'll probably name it `bundleTestsToRun`. What this is doing is filtering out the tests from the bundle that aren't included in the `testCasesToRun` parameter passed in.